src/js/user.js

```diff
@@ -8,6 +8,30 @@ function UserManager(auth, server) {
     var api = new Gogs(server + '/api/v1'),
         tokenStub = {name: 'btt-writer-desktop'};
 
+    const MAX_PAGE_SIZE = 50
+
+    const fetchRepoRecursively = function (uid, query, limit, page, repoList) {
```

Comment:
I appreciate the elegance of this recursive approach, but I'm concerned that users with many pages of repos (e.g. Staging or DCS-Mirror) could use a lot of memory on the stack or even cause it to overflow.

I recommend converting this from a recursive strategy to an iterative one.  For example, a loop that keeps calling `api.searchRepos()` until it returns an array of length zero.

Reply:
I got into an error earlier (unexpected token function) when trying to use a while loop with async/await function, because we're working with Promise. I just found the error was that because in this old version of electron/node, the async/await function wasn't born yet.

So we will probably need to merge the upgrade platform (#38) before I get the script working with that async/await